plugins/cindy3d/src/js/PrimitiveRenderer.js

```diff
@@ -219,9 +229,22 @@ PrimitiveRenderer.prototype.addPrimitive = function(attributes) {
     throw new GlError("Wrong number of attributes given");
   if (this.count == this.capacity) {
     let c = this.capacity*2, nd, nda, ndi, i, j, k, o, e = this.elements;
+    if (c > this.maxCount) {
+      if (this.capacity < this.maxCount) {
+        c = this.maxCount;
+      } else {
+        this.indexByteCount = 4;
+        this.itemTotalByteCount = this.itemAttribByteCount +
+          this.numElements * this.indexByteCount;
+        this.maxCount = Math.floor((1 << 31) / this.itemAttribByteCount);
```

Comment:
I think `Math.pow` is the appropriate thing to use, with `(1<<16)*(1<<16)` as an alternative I'd find agreeable as well.